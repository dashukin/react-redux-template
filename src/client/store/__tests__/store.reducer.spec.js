import { combineReducers } from 'redux';
import {
  reducers,
  createCombinedReducers,
} from '../store.reducer';

jest.mock('redux');

describe('store.reducer', () => {
  describe('reducers', () => {
    it('should match snapshot', () => {
      expect(reducers).toMatchSnapshot();
    });
  });

  describe('createCombinedReducers', () => {
    it('should combine store reducers with any extra reducers passed', () => {
      const mockExtraReducer1 = jest.fn();
      const mockExtraReducer2 = jest.fn();
      const mockExtraReducersMap = { mockExtraReducer1, mockExtraReducer2 };
      const expectedReducersArgument = {
        ...reducers,
        ...mockExtraReducersMap,
      };
      const expectedCombineReducersResult = jest.fn();
      combineReducers.mockReturnValue(expectedCombineReducersResult);

      const combinedReducers = createCombinedReducers(mockExtraReducersMap);

      expect(combineReducers).toHaveBeenCalledWith(expectedReducersArgument);
      expect(combinedReducers).toEqual(expectedCombineReducersResult);
    });
  });
});
