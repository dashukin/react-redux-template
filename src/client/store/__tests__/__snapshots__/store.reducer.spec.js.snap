// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store.reducer reducers should match snapshot 1`] = `
Object {
  "db": [Function],
  "example": [Function],
  "page": [Function],
}
`;
